mobile/src/main/java/org/openhab/habdroid/util/Util.java

```diff
@@ -281,6 +287,34 @@ public static void applyAuthentication(WebView webView, Connection connection, S
             webView.setHttpAuthUsernamePassword(Util.getHostFromUrl(url), "",
                     connection.getUsername(), connection.getPassword());
         }
+
+        webView.getSettings().setDomStorageEnabled(true);
+        webView.getSettings().setJavaScriptEnabled(true);
+        webView.getSettings().setSupportMultipleWindows(true);
+
+        webView.setWebChromeClient(new WebChromeClient() {
+            @Override
+            public boolean onCreateWindow(WebView view, boolean dialog, boolean userGesture,
+                    Message resultMsg) {
+                Message href = view.getHandler().obtainMessage();
+                view.requestFocusNodeHref(href);
+                String url = href.getData().getString("url");
+                if (url != null) {
+                    openInBrowser(url, view.getContext());
+                }
+                return false;
+            }
+        });
+    }
+
+    public static void openInBrowser(String url, Context context) {
+        Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(url));
```

Comment:
IMO you should check here for `!TextUtils.isEmpty(url)` and for url being a valid url. Then add `@Nullable String url` in the method args and remove the null check in `onCreateWindow()`.